servers/dsub/jobs/controllers/aggregation_controller.py

```diff
@@ -24,47 +19,110 @@ def get_job_aggregations(timeFrame, projectId=None):
     Returns:
         AggregationResponse: Response contains aggregation of jobs
     """
-    # temporary fake data for testing front-end.
-    statusCount1 = StatusCount(status='Succeeded', count=10)
-
-    statusCount2 = StatusCount(status='Failed', count=2)
-
-    summaryCounts = StatusCounts(counts=[statusCount1, statusCount2])
-
-    entry1 = AggregationEntry(
-        label='labelValue1',
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=2),
-            StatusCount(status='Failed', count=1)
-        ]))
-
-    entry2 = AggregationEntry(
-        label='labelValue2',
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=4),
-            StatusCount(status='Failed', count=6)
-        ]))
-
-    userEntry1 = AggregationEntry(
-        label='circleci',
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=3),
-            StatusCount(status='Failed', count=7)
-        ]))
-
-    userEntry2 = AggregationEntry(
-        label='otherUser',
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=7),
-            StatusCount(status='Failed', count=4)
-        ]))
-
-    ownerAggregation = Aggregation(
-        name='AnotherLabel', key='job-id', entries=[entry1, entry2])
-
-    projectAggregation = Aggregation(
-        name='User', key='userId', entries=[userEntry1, userEntry2])
+    window_min = time_frame.time_frame_to_start_time(timeFrame)
+    provider = providers.get_provider(_provider_type(), projectId,
+                                      _auth_token())
+
+    jobs = jobs_generator.generate_jobs_by_window(provider, projectId,
+                                                  window_min)
+
+    total_summary = {}
+    user_summary = {}
+    job_name_summary = {}
+    label_summaries = {}
+    label_freq = {}
+
+    for job in jobs:
+        _count_total_summary(job, total_summary)
+        _count_for_key(job, user_summary, lambda j: j.extensions.user_id)
+        _count_for_key(job, job_name_summary, lambda j: j.name)
+        _count_top_labels(label_summaries, job, label_freq)
+
+    aggregations = [
+        _to_aggregation('User Id', 'userId', user_summary),
+        _to_aggregation('Job Name', 'name', job_name_summary)
+    ]
+
+    _to_top_labels_aggregations(label_freq, label_summaries, aggregations)
```

Comment:
In general I would not recommend the pattern of passing in a value and mutating it if it is avoidable. On the above count_ methods, this is sort of acceptable since you are accumulating things as you iterate. On this line it would be simpler to have the function return a list of aggregations.

```
aggregations = [
 ...
] + _to_top_labels_aggregations(...)
```